### ISSUE TITLE
Recursive splat expansion error when mounting Shared::Field

### ISSUE THREAD

**Issue by matthewmcgarvey** [luckyframework/lucky#1299]
**Describe the bug**
When creating a form, we attempted to move a `Shared::Field` that did not pass a block (it relied on the default text_input override) to the first input above the email field. We started getting a compilation error like:
```
mount Shared::FieldErrors, attribute
^----
Error: recursive splat expansion: String, Avram::PermittedAttribute(String | Nil), ...
```

It only happens when the first field is using the override that provides a block.

**To Reproduce**

Here is a branch that has the failing code in a spec. I did try to simplify it, but I'm sure it could be made more straight forward.  https://github.com/matthewmcgarvey/lucky/tree/recursive-splat-expansion

The diff:
https://github.com/luckyframework/lucky/compare/master...matthewmcgarvey:recursive-splat-expansion

**Expected behavior**

Any ordering of mounted fields should work.

**Versions (please complete the following information):**
 - Lucky version (check in shard.lock): v0.24.0
 - Crystal version (`crystal --version`): v0.35.1
 - OS: MacOS

**Comment by matthewmcgarvey** [comment]
So the problem goes away if I move the methods into the component or I inline the `mount` code. There is a bug on crystal for this error message, could I have stumbled onto this problem? https://github.com/crystal-lang/crystal/issues/7164

**Comment by matthewmcgarvey** [comment]
Verified to still be an issue on Crystal v0.36.1

**Comment by snadon** [comment]
I was able to remove my workaround with Lucky 0.29 and Crystal 1.3.2. Not sure when the problem went away though.

**Comment by matthewmcgarvey** [comment]
<img width="634" alt="Screen Shot 2022-01-27 at 8 19 56 PM" src="https://user-images.githubusercontent.com/17329408/151475862-6588ddf4-52a3-4ca5-98d3-47100b109f4f.png">

Verified to still be an issue on Crystal v1.3.0 😢

**Comment by asterite** [comment]
What's the full error with --error-trace?

**Comment by matthewmcgarvey** [comment]
<img width="698" alt="Screen Shot 2022-01-29 at 1 05 52 PM" src="https://user-images.githubusercontent.com/17329408/151674168-4a8ef26a-859a-4e32-becf-a35bf781a647.png">

**Comment by flipsasser** [comment]
I'm still bedeviled by this in 2022. Anyone have an ideas?

**Comment by jwoertink** [comment]
@flipsasser are you using Crystal 1.6.2? Are you able to give Crystal 1.7.0-dev a try to see if that gives you the same issue?

**Comment by flipsasser** [comment]
@jwoertink upgrading solved it! I was still on 1.4.2. My fault. Thanks for the tip!

**Comment by jwoertink** [comment]
Sweet! Yeah, I think this is fixed now with the later version of Crystal... Though, we say we support 1.4.0 or later... maybe that needs to be bumped up :thinking:
